pkg/streamingpromql/operators/functions/range_vectors.go

```diff
@@ -369,3 +369,62 @@ func avgHistograms(head, tail []promql.HPoint) (*histogram.FloatHistogram, error
 
 	return avgSoFar, nil
 }
+
+var Deriv = FunctionOverRangeVectorDefinition{
+	SeriesMetadataFunction: DropSeriesName,
+	StepFunc:               deriv,
+}
+
+func deriv(step types.RangeVectorStepData, _ float64, _ types.EmitAnnotationFunc) (float64, bool, *histogram.FloatHistogram, error) {
+	head, tail := step.Floats.UnsafePoints(step.RangeEnd)
+
+	if len(head) < 2 && len(tail) == 0 {
+		return 0, false, nil, nil
+	}
+
+	slope, _ := linearRegression(append(head, tail...), head[0].T)
```

Comment:
Rather than appending `head` and `tail` like this, what if we pass both to `linearRegression` and modify it to support two slices of points?